airflow/providers/apache/kafka/operators/consume.py

```diff
@@ -0,0 +1,184 @@
+    :param apply_function_batch: The function that should be applied to a batch of messages fetched. Can not
+        be used with `apply_function`. Intended for transactional workloads where an expensive task might
+        be called before or after operations on the messages are taken.
+    :param apply_function_args: Additional arguments that should be applied to the callable, defaults to None
+    :param apply_function_kwargs: Additional key word arguments that should be applied to the callable
+        defaults to None
+    :param commit_cadence: When consumers should commit offsets ("never", "end_of_batch","end_of_operator"),
+        defaults to "end_of_operator";
+        if end_of_operator, the commit() is called based on the max_messages arg. Commits are made after the
+        operator has processed the apply_function method for the maximum messages in the operator.
+        if end_of_batch, the commit() is called based on the max_batch_size arg. Commits are made after each
+        batch has processed by the apply_function method for all messages in the batch.
+        if never,  close() is called without calling the commit() method.
+    :param max_messages: The maximum total number of messages an operator should read from Kafka,
+        defaults to None
+    :param max_batch_size: The maximum number of messages a consumer should read when polling,
+        defaults to 1000
+    :param poll_timeout: How long the Kafka consumer should wait before determining no more messages are
+        available, defaults to 60
+
+    .. seealso::
+        For more information on how to use this operator, take a look at the guide:
+        :ref:`howto/operator:ConsumeFromTopicOperator`
+    """
+
+    BLUE = "#ffefeb"
+    ui_color = BLUE
+    template_fields = (
+        "topics",
+        "apply_function",
+        "apply_function_args",
+        "apply_function_kwargs",
+        "kafka_config_id",
+    )
+
+    def __init__(
+        self,
+        topics: str | Sequence[str],
+        kafka_config_id: str = "kafka_default",
+        apply_function: Callable[..., Any] | str | None = None,
+        apply_function_batch: Callable[..., Any] | str | None = None,
+        apply_function_args: Sequence[Any] | None = None,
+        apply_function_kwargs: dict[Any, Any] | None = None,
+        commit_cadence: str | None = "end_of_operator",
+        max_messages: int | None = None,
+        max_batch_size: int = 1000,
+        poll_timeout: float = 60,
+        **kwargs: Any,
+    ) -> None:
+
+        super().__init__(**kwargs)
+
+        self.topics = topics
+        self.apply_function = apply_function
+        self.apply_function_batch = apply_function_batch
+        self.apply_function_args = apply_function_args or ()
+        self.apply_function_kwargs = apply_function_kwargs or {}
+        self.kafka_config_id = kafka_config_id
+        self.commit_cadence = commit_cadence
+        self.max_messages = max_messages or True
```

Comment:
what should happen when the user provide 0?

Reply:
```python
>>> y = 0 or True
>>> y
True
``` 
^that.

Reply:
I understand what happen with the current code, my question was more about what is supposed to happen.

> Functionally this means "read to the end of the log and exit when kafka has no more messages".

Can you please add it to the doc to avoid the confusion?